src/src/Helpers/Hardware.cpp

```diff
@@ -163,9 +163,15 @@ void hardwareInit()
 #endif // ifdef FEATURE_SD
 }
 
+bool isI2CEnabled() {
+  return ((Settings.Pin_i2c_sda != -1) &&
+          (Settings.Pin_i2c_scl != -1) &&
+          (Settings.I2C_clockSpeed > 0));
+}
```

Comment:
Uses only members of Settings, so why not make it a member function of Settings?
Next to the helper functions there to collect other protocol specific info.  (e.g. SPI pins)